server/monitor-types/snmp.js

```diff
@@ -0,0 +1,107 @@
+const { MonitorType } = require("./monitor-type");
+const { UP, DOWN, log } = require("../../src/util");
+const snmp = require("net-snmp");
+
+class SNMPMonitorType extends MonitorType {
+    name = "snmp";
+
+    /**
+     * Checks the SNMP value against the condition and control value.
+     * @param {object} monitor The monitor object associated with the check.
+     * @param {object} heartbeat The heartbeat object to update.
+     * @param {object} _server Unused server object.
+     */
+    async check(monitor, heartbeat, _server) {
+
+        log.debug("monitor", `SNMP: Community String: ${monitor.snmpCommunityString}`);
+        log.debug("monitor", `SNMP: OID: ${monitor.snmpOid}`);
+        log.debug("monitor", `SNMP: Version: ${monitor.snmpVersion}`);
+        log.debug("monitor", `SNMP: Condition: ${monitor.snmpCondition}`);
+        log.debug("monitor", `SNMP: Control Value: ${monitor.snmpControlValue}`);
+
+        const options = {
+            port: monitor.port || '161',
+            retries: monitor.maxretries,
+            timeout: 1000,
```

Comment:
please configure the timeout like the other monitors

```suggestion
            timeout: monitor.interval * 1000 * 0.8,
```

or (with the frontend changes)

```suggestion
            timeout: monitor.timeout,
```

Reply:
Resolved with f059d54349d0b71c297b504e167fd3e7d5844c47.